platform_umbrella/apps/common_core/lib/common_core/resources/karpenter_pools.ex

```diff
@@ -161,4 +143,21 @@ defmodule CommonCore.Resources.KarpenterPools do
     |> B.name("amd-gpu")
     |> B.spec(spec)
   end
+
+  defp build_nodeclass_spec(family, alias, role, cluster_name) do
+    %{
+      "amiFamily" => family,
+      "amiSelectorTerms" => [%{"alias" => alias}],
+      "role" => role,
+      "securityGroupSelectorTerms" => [%{"tags" => %{"karpenter.sh/discovery" => cluster_name}}],
+      "subnetSelectorTerms" => [%{"tags" => %{"karpenter.sh/discovery" => cluster_name}}],
+      "metadataOptions" => %{"httpPutResponseHopLimit" => 2},
```

Comment:
I refactored this spec generation but setting this hop limit to 2 is the fix.

Reply:
Please comment on why this is needed and how it fixes it.